README.md
# google-material-color

[![GitHub version](https://badge.fury.io/gh/danlevan%2Fgoogle-material-color.svg)](http://badge.fury.io/gh/danlevan%2Fgoogle-material-color)
[![npm version](https://badge.fury.io/js/google-material-color.svg)](http://badge.fury.io/js/google-material-color) [![Bower version](https://badge.fury.io/bo/google-material-color.svg)](http://badge.fury.io/bo/google-material-color)

[Google material color](http://www.google.com/design/spec/style/color.html#color-color-palette) implementation for Sass, Less, Stylus, CSS and JS.

> This color palette comprises primary and accent colors that can be used for illustration or to develop your brand colors. They’ve been designed to work harmoniously with each other.
([http://www.google.com/design/spec/style/color.html](http://www.google.com/design/spec/style/color.html#color-color-palette)).

See the [demo](http://danlevan.github.io/google-material-color/test/) generated by the tests.

## Installation

* Download the files you need from the [dist](dist) directory:
  * Stylus:  [palette.styl](http://danlevan.github.io/google-material-color/dist/palette.styl)
  * Less:  [palette.less](http://danlevan.github.io/google-material-color/dist/palette.less)
  * Sass (SCSS) 3.3+:  [palette.scss](http://danlevan.github.io/google-material-color/dist/palette.scss)
  * CSS:  [palette.css](http://danlevan.github.io/google-material-color/dist/palette.css)
  * CSS variable:  [palette.css](http://danlevan.github.io/google-material-color/dist/palette-var.css)
  * JS:  [palette.js](http://danlevan.github.io/google-material-color/dist/palette.js) supports (AMD, node, browser)
* NPM: `$ npm install google-material-color --save`
* Bower: `$ bower install google-material-color --save`
* Git: `$ git clone git://github.com/danlevan/google-material-color.git`

## Colors

Details can be found on the [google design specs website](http://www.google.com/design/spec/style/color.html#color-color-palette).

**Colors**
> Red, Pink, Purple, Deep Purple, Indigo, Blue, Light Blue, Cyan, Teal, Green, Light Green, Lime, Yellow, Amber, Orange, Deep Orange, Brown, Grey, Blue Grey

**Shades** (note that not all color have all the shades)
> 50, 100, 200, 300, 400, 500, 600, 700, 800, 900, 1000, A100, A200, A400, A700

**Text Colors**
> White, Black

**Text Shades**
> Primary, Secondary, Icons, Disabled, Hint, Dividers


## Usage

### Stylus

Import [palette.styl](http://danlevan.github.io/google-material-color/dist/palette.styl).

Call the function `palette('[color]', '[shade]')`.

> The shade is optional (500 is the default shade).

Call the function `paletteText('[color]', '[shade]', '[textShade]')`.

> The shade is optional (500 is the default shade). The textShade is optional (Primary is the default textShade).

`example.styl`

```stylus
@import 'palette'
...

.my-div
  background-color palette('Light Blue', '700')
  color paletteText('Red', '700', 'Secondary')

.my-default-div
  background-color palette('Light Blue') // default shade is 500
  color paletteText('Red') // default text shade is Primary
```

> If you need direct access to the variables, you can access it like `$palette['Light Blue']['500']` and `$palette[$paletteText['Light Blue']['500']]['Primary']`

### Sass (SCSS)

Import [palette.scss](http://danlevan.github.io/google-material-color/dist/palette.scss).

Call the function `palette([color], [shade])`.

> The shade is optional (500 is the default shade).

Call the function `paletteText([color], [shade], [textShade])`.

> The shade is optional (500 is the default shade). The textShade is optional (Primary is the default textShade).

`example.scss`

```scss
@import 'palette';
...

.my-div {
  background-color: palette(Light Blue, 700);
  color: paletteText(Light Blue, 700, Secondary);
}

.my-default-div {
  background-color: palette(Red); // default shade is 500
  color: paletteText(Light Blue); // default text shade is Primary
}
```

> If you need direct access to the variables, you can access it through a map like `$colorMap: map-get($palette, Light Blue); $color: map-get($colorMap, 700);` and `$colorMap: map-get($paletteText, Light Blue); $textColor: map-get($colorMap, 700); $colorMap: map-get($palette, $textColor); $color: map-get($colorMap, Secondary);`.

### Less

Import [palette.less](http://danlevan.github.io/google-material-color/dist/palette.less).

Call the mixin `.palette([color], [shade])`.

> The shade is optional (500 is the default shade).

Call the mixin `.palette-text([color], [shade], [textShade])`.

> The shade is optional (500 is the default shade). The textShade is optional (Primary is the default textShade).

`example.scss`

```less
@import 'palette';
...

.my-div {
  .palette('Light Blue', '700');
  background-color: @palette;

  // get @palette in its own scope otherwise @palette will be overwritten
  &{
    .palette('Red'); // default shade is 500
    color: @palette;
  }
}

```

If you need access to the variables, you can access it through variable like `@palette-Light-Blue-500` and `@colorVar: %('palette-%s-Secondary', @palette-text-Light-Blue-500); @color: @@colorVar`


### CSS

Include [palette.css](http://danlevan.github.io/google-material-color/dist/palette.css) in your html. The CSS class follows the pattern `palette-[color]-[shade]-[textShade]`, (spaces replaced by `-`). 

The CSS provides colors for the background and text

* `background-color`: by adding the `bg` class to the element
* `color`: will be White or Black if adding the `bg` class to the element; will be the color setted if adding the `text` class (does not support textShade) to the element



`example.html`

```html
<link href='palette.css' rel='stylesheet' type='text/css'>
...

<div class="palette-Light-Blue-700-Secondary bg">
  The background is Light Blue
  The text is White
</div>

<div class="palette-Light-Blue-700 text">
  The text is Light Blue
</div>
```

### CSS variable

If you're using cssnext, this project also includes a variable files for css. Read more here [Custom properties](http://cssnext.io/features/#custom-properties-var).

Include [palette.css](http://danlevan.github.io/google-material-color/dist/palette-var.css) in your html.
```css
:root {
  --palette-Red-50: #FFEBEE
  --palette-Red-100: #FFCDD2
  --palette-Red-200: #EF9A9A
  ...
  --palette-Red-Primary: #ffffff;
  --palette-Red-Secondary: rgba(255,255,255,0.7);
  ...
  --palette-Red-50-Primary: #ffffff;
  --palette-Red-50-Secondary: rgba(255,255,255,0.7);
  ...
```

### JS

You can import the [palette.js](http://danlevan.github.io/google-material-color/dist/palette.js) file in Node.js, and AMD module or the browser.

`example.html`

```javascript
<script src='../dist/palette.js'></script>
...

<script>
  document.getElementById('my-div')
    .style['background-color'] = palette.get('Light Blue', '700');
    .style['color'] = palette.getText('Light Blue', '700', 'Secondary');
</script>
```


## Issues

Search or open new issues [here][issues].

If you would like to suggest improvements or other libraries, you can also open an issue.

## Contributing

See the [contributing](CONTRIBUTING.md) doc

## Release History
See the [changelog](CHANGELOG.md)

## License

Licensed under [MIT](LICENSE).

Google material design [Terms](http://www.google.com/design/spec/style/color.html).


[issues]: https://github.com/danlevan/google-material-color/issues "GitHub Issues for Less.js"
[download]: https://github.com/less/less.js/zipball/master "Download Less.js"


CHANGELOG.md
# 1.3.0

2016-11-15

- Add css text

# 1.2.6

2015-08-07

- Add css variables

# 1.2.3

2015-08-07

- Fix documentation for less (issue #7)

# 1.2.2

2015-03-20

- Add random function in the js version


# 1.2.0

2014-11-21

- Update the color (update from Google docs)

# 1.1.0

2014-11-20

- Added text color (alpha) for black and white

# 1.0.0

2014-11-14

- Fixed black and white

# 0.0.2

2014-11-11

- Add the header to the files

# 0.0.1

2014-11-10

 - stylus implementation
 - sass implementation
 - less implementation
 - css implementation
 - js implementation


gulpfile.js
var _ = require('lodash');
var clean = require('gulp-clean');
var colors = require('./src/colors');
var compass = require('gulp-compass')
var gulp = require('gulp');
var header = require('gulp-header');
var less = require('gulp-less');
var path = require('path');
var packageConfig = require('./package.json');
var rename = require('gulp-rename');
var sass = require('gulp-ruby-sass');
var stylus = require('gulp-stylus');
var sync = require('gulp-config-sync');
var template = require('gulp-template');
var util = require('util');

var paths = {
  destination: 'dist',
  test: 'test',

  stylus: 'palette.styl',
  sass: 'palette.scss',
  less: 'palette.less',
  css: 'palette.css',
  cssVar: 'palette.var.css',
  js: 'palette.js',
};

var banner = '' +
  '/**\n' +
  ' * <%= name %> v<%= version %>\n' +
  ' * <%= homepage %>\n' +
  ' */\n';

/**
 * Generates all the plugins
 */
gulp.task('default', [/*'sync', */'clean-dist', 'stylus', 'sass', 'less', 'css', 'css-var', 'js']);

/**
 * Generates the test files: compile lib to css
 */
gulp.task('test', ['clean-test', 'test-page', 'test-stylus', 'test-sass', 'test-less', 'css', 'css-var', 'js']);

////////////////////////////////////////////////////////////////////////////////
// CSS
////////////////////////////////////////////////////////////////////////////////

/**
 * Generates the css files
 */
gulp.task('css', ['clean-dist', 'clean-test'], function() {
  return gulp.src('src/templates/css')
    .pipe(template({ colors: colors, }))
    .pipe(rename(paths.css))
    .pipe(header(banner, packageConfig))
    .pipe(gulp.dest(paths.destination));
});

/**
 * Generates the css-var files
 */
gulp.task('css-var', ['clean-dist', 'clean-test'], function() {
  return gulp.src('src/templates/css-var')
    .pipe(template({ colors: colors, }))
    .pipe(rename(paths.cssVar))
    .pipe(header(banner, packageConfig))
    .pipe(gulp.dest(paths.destination));
});

////////////////////////////////////////////////////////////////////////////////
// JSON
////////////////////////////////////////////////////////////////////////////////

/**
 * Generates the js files
 */
gulp.task('js', ['clean-dist', 'clean-test'], function() {
  return gulp.src('src/templates/js')
    .pipe(template({ colors: colors, }))
    .pipe(rename(paths.js))
    .pipe(header(banner, packageConfig))
    .pipe(gulp.dest(paths.destination));
});

////////////////////////////////////////////////////////////////////////////////
// STYLUS
////////////////////////////////////////////////////////////////////////////////

/**
 * Generates the stylus files
 */
gulp.task('stylus', ['clean-dist', 'clean-test'], function() {
  return gulp.src('src/templates/stylus')
    .pipe(template({ colors: colors, }))
    .pipe(rename(paths.stylus))
    .pipe(header(banner, packageConfig))
    .pipe(gulp.dest(paths.destination));
});

/**
 * Stylus css for test
 */
gulp.task('test-stylus', ['stylus'], function() {
  gulp.src('src/templates/test-stylus')
    .pipe(template({
      path: path.join('../../', paths.destination, paths.stylus),
      colors: colors,
    }))
    .pipe(stylus())
    .pipe(rename('stylus.css'))
    .pipe(gulp.dest(paths.test));
});

////////////////////////////////////////////////////////////////////////////////
// SASS
////////////////////////////////////////////////////////////////////////////////

/**
 * Generates the sass files
 */
gulp.task('sass', ['clean-dist', 'clean-test'], function() {
  return gulp.src('src/templates/scss')
    .pipe(template({ colors: colors, }))
    .pipe(rename(paths.sass))
    .pipe(header(banner, packageConfig))
    .pipe(gulp.dest(paths.destination));
});

/**
 * Sass css for test
 */
gulp.task('test-sass', ['sass', 'test-sass-template'], function() {
  gulp.src(path.join(paths.test, '*.scss'))
    .pipe(compass({
      css: paths.test,
      sass: paths.test,
      import_path: paths.destination,
    }))
    .pipe(gulp.dest(paths.test));
})

gulp.task('test-sass-template', function() {
  return gulp.src('src/templates/test-scss')
    .pipe(template({
      path: paths.sass,
      colors: colors,
    }))
    .pipe(rename('sass.scss'))
    .pipe(gulp.dest(paths.test));
});

////////////////////////////////////////////////////////////////////////////////
// LESS
////////////////////////////////////////////////////////////////////////////////

/**
 * Generates the less files
 */
gulp.task('less', ['clean-dist', 'clean-test'], function() {
  return gulp.src('src/templates/less')
    .pipe(template({ colors: colors, }))
    .pipe(rename(paths.less))
    .pipe(header(banner, packageConfig))
    .pipe(gulp.dest(paths.destination));
});

gulp.task('test-less', ['less', 'test-less-template'], function () {
  gulp.src(path.join(paths.test, '*.less'))
    .pipe(less({
      paths: [ path.join(__dirname, paths.destination) ],
    }))
    .pipe(rename('less.css'))
    .pipe(gulp.dest(paths.test));
});

gulp.task('test-less-template', function() {
  return gulp.src('src/templates/test-less')
    .pipe(template({
      path: path.join('../', paths.destination, paths.less),
      colors: colors,
    }))
    .pipe(rename('less.less'))
    .pipe(gulp.dest(paths.test));
});

////////////////////////////////////////////////////////////////////////////////
// TESTS
////////////////////////////////////////////////////////////////////////////////

/**
 * Test page html
 */
gulp.task('test-page', ['clean-dist', 'clean-test'], function() {
  gulp.src('src/templates/test-page')
    .pipe(template({
      path: {
        css: path.join('../', paths.destination, paths.css),
        js: path.join('../', paths.destination, paths.js),
      },
      colors: colors,
    }))
    .pipe(rename('index.html'))
    .pipe(gulp.dest(paths.test));
});

////////////////////////////////////////////////////////////////////////////////
// CLEAN
////////////////////////////////////////////////////////////////////////////////

/**
 * Cleans the distribution folder
 */
gulp.task('clean-dist', function() {
  return gulp.src(paths.destination, {read: false})
    .pipe(clean());
});

/**
 * Cleans the test
 */
gulp.task('clean-test', function() {
  return gulp.src(paths.test, {read: false})
    .pipe(clean());
});

////////////////////////////////////////////////////////////////////////////////
// VERSIONING
////////////////////////////////////////////////////////////////////////////////

gulp.task('sync', function() {
  return gulp.src('bower.json')
    .pipe(sync())
    .pipe(gulp.dest('.'));
});

////////////////////////////////////////////////////////////////////////////////
// CLEAN
////////////////////////////////////////////////////////////////////////////////

/**
 * Watch dev
 */
gulp.task('watch', ['default', 'test'], function () {
  gulp.watch('src/templates/*', ['default', 'test']);
});


src/colors.js
// Google style palette (http://www.google.com/design/spec/style/'color'.html': '#'color'-ui-'color'-palette)

module.exports = {

  'Red': {
    '50': { 'color': '#FFEBEE', 'text': 'Black' },
    '100': { 'color': '#FFCDD2', 'text': 'Black' },
    '200': { 'color': '#EF9A9A', 'text': 'Black' },
    '300': { 'color': '#E57373', 'text': 'Black' },
    '400': { 'color': '#EF5350', 'text': 'White' },
    '500': { 'color': '#F44336', 'text': 'White' },
    '600': { 'color': '#E53935', 'text': 'White' },
    '700': { 'color': '#D32F2F', 'text': 'White' },
    '800': { 'color': '#C62828', 'text': 'White' },
    '900': { 'color': '#B71C1C', 'text': 'White' },
    'A100': { 'color': '#FF8A80', 'text': 'Black' },
    'A200': { 'color': '#FF5252', 'text': 'White' },
    'A400': { 'color': '#FF1744', 'text': 'White' },
    'A700': { 'color': '#D50000', 'text': 'White' }
  },
  'Pink': {
    '50': { 'color': '#FCE4EC', 'text': 'Black' },
    '100': { 'color': '#F8BBD0', 'text': 'Black' },
    '200': { 'color': '#F48FB1', 'text': 'Black' },
    '300': { 'color': '#F06292', 'text': 'White' },
    '400': { 'color': '#EC407A', 'text': 'White' },
    '500': { 'color': '#E91E63', 'text': 'White' },
    '600': { 'color': '#D81B60', 'text': 'White' },
    '700': { 'color': '#C2185B', 'text': 'White' },
    '800': { 'color': '#AD1457', 'text': 'White' },
    '900': { 'color': '#880E4F', 'text': 'White' },
    'A100': { 'color': '#FF80AB', 'text': 'Black' },
    'A200': { 'color': '#FF4081', 'text': 'White' },
    'A400': { 'color': '#F50057', 'text': 'White' },
    'A700': { 'color': '#C51162', 'text': 'White' }
  },
  'Purple': {
    '50': { 'color': '#F3E5F5', 'text': 'Black' },
    '100': { 'color': '#E1BEE7', 'text': 'Black' },
    '200': { 'color': '#CE93D8', 'text': 'Black' },
    '300': { 'color': '#BA68C8', 'text': 'White' },
    '400': { 'color': '#AB47BC', 'text': 'White' },
    '500': { 'color': '#9C27B0', 'text': 'White' },
    '600': { 'color': '#8E24AA', 'text': 'White' },
    '700': { 'color': '#7B1FA2', 'text': 'White' },
    '800': { 'color': '#6A1B9A', 'text': 'White' },
    '900': { 'color': '#4A148C', 'text': 'White' },
    'A100': { 'color': '#EA80FC', 'text': 'Black' },
    'A200': { 'color': '#E040FB', 'text': 'White' },
    'A400': { 'color': '#D500F9', 'text': 'White' },
    'A700': { 'color': '#AA00FF', 'text': 'White' }
  },
  'Deep Purple': {
    '50': { 'color': '#EDE7F6', 'text': 'Black' },
    '100': { 'color': '#D1C4E9', 'text': 'Black' },
    '200': { 'color': '#B39DDB', 'text': 'Black' },
    '300': { 'color': '#9575CD', 'text': 'White' },
    '400': { 'color': '#7E57C2', 'text': 'White' },
    '500': { 'color': '#673AB7', 'text': 'White' },
    '600': { 'color': '#5E35B1', 'text': 'White' },
    '700': { 'color': '#512DA8', 'text': 'White' },
    '800': { 'color': '#4527A0', 'text': 'White' },
    '900': { 'color': '#311B92', 'text': 'White' },
    'A100': { 'color': '#B388FF', 'text': 'Black' },
    'A200': { 'color': '#7C4DFF', 'text': 'White' },
    'A400': { 'color': '#651FFF', 'text': 'White' },
    'A700': { 'color': '#6200EA', 'text': 'White' }
  },
  'Indigo': {
    '50': { 'color': '#E8EAF6', 'text': 'Black' },
    '100': { 'color': '#C5CAE9', 'text': 'Black' },
    '200': { 'color': '#9FA8DA', 'text': 'Black' },
    '300': { 'color': '#7986CB', 'text': 'White' },
    '400': { 'color': '#5C6BC0', 'text': 'White' },
    '500': { 'color': '#3F51B5', 'text': 'White' },
    '600': { 'color': '#3949AB', 'text': 'White' },
    '700': { 'color': '#303F9F', 'text': 'White' },
    '800': { 'color': '#283593', 'text': 'White' },
    '900': { 'color': '#1A237E', 'text': 'White' },
    'A100': { 'color': '#8C9EFF', 'text': 'Black' },
    'A200': { 'color': '#536DFE', 'text': 'White' },
    'A400': { 'color': '#3D5AFE', 'text': 'White' },
    'A700': { 'color': '#304FFE', 'text': 'White' }
  },
  'Blue': {
    '50': { 'color': '#E3F2FD', 'text': 'Black' },
    '100': { 'color': '#BBDEFB', 'text': 'Black' },
    '200': { 'color': '#90CAF9', 'text': 'Black' },
    '300': { 'color': '#64B5F6', 'text': 'Black' },
    '400': { 'color': '#42A5F5', 'text': 'Black' },
    '500': { 'color': '#2196F3', 'text': 'White' },
    '600': { 'color': '#1E88E5', 'text': 'White' },
    '700': { 'color': '#1976D2', 'text': 'White' },
    '800': { 'color': '#1565C0', 'text': 'White' },
    '900': { 'color': '#0D47A1', 'text': 'White' },
    'A100': { 'color': '#82B1FF', 'text': 'Black' },
    'A200': { 'color': '#448AFF', 'text': 'White' },
    'A400': { 'color': '#2979FF', 'text': 'White' },
    'A700': { 'color': '#2962FF', 'text': 'White' }
  },
  'Light Blue': {
    '50': { 'color': '#E1F5FE', 'text': 'Black' },
    '100': { 'color': '#B3E5FC', 'text': 'Black' },
    '200': { 'color': '#81D4FA', 'text': 'Black' },
    '300': { 'color': '#4FC3F7', 'text': 'Black' },
    '400': { 'color': '#29B6F6', 'text': 'Black' },
    '500': { 'color': '#03A9F4', 'text': 'Black' },
    '600': { 'color': '#039BE5', 'text': 'White' },
    '700': { 'color': '#0288D1', 'text': 'White' },
    '800': { 'color': '#0277BD', 'text': 'White' },
    '900': { 'color': '#01579B', 'text': 'White' },
    'A100': { 'color': '#80D8FF', 'text': 'Black' },
    'A200': { 'color': '#40C4FF', 'text': 'Black' },
    'A400': { 'color': '#00B0FF', 'text': 'Black' },
    'A700': { 'color': '#0091EA', 'text': 'White' }
  },
  'Cyan': {
    '50': { 'color': '#E0F7FA', 'text': 'Black' },
    '100': { 'color': '#B2EBF2', 'text': 'Black' },
    '200': { 'color': '#80DEEA', 'text': 'Black' },
    '300': { 'color': '#4DD0E1', 'text': 'Black' },
    '400': { 'color': '#26C6DA', 'text': 'Black' },
    '500': { 'color': '#00BCD4', 'text': 'Black' },
    '600': { 'color': '#00ACC1', 'text': 'Black' },
    '700': { 'color': '#0097A7', 'text': 'White' },
    '800': { 'color': '#00838F', 'text': 'White' },
    '900': { 'color': '#006064', 'text': 'White' },
    'A100': { 'color': '#84FFFF', 'text': 'Black' },
    'A200': { 'color': '#18FFFF', 'text': 'Black' },
    'A400': { 'color': '#00E5FF', 'text': 'Black' },
    'A700': { 'color': '#00B8D4', 'text': 'Black' }
  },
  'Teal': {
    '50': { 'color': '#E0F2F1', 'text': 'Black' },
    '100': { 'color': '#B2DFDB', 'text': 'Black' },
    '200': { 'color': '#80CBC4', 'text': 'Black' },
    '300': { 'color': '#4DB6AC', 'text': 'Black' },
    '400': { 'color': '#26A69A', 'text': 'Black' },
    '500': { 'color': '#009688', 'text': 'White' },
    '600': { 'color': '#00897B', 'text': 'White' },
    '700': { 'color': '#00796B', 'text': 'White' },
    '800': { 'color': '#00695C', 'text': 'White' },
    '900': { 'color': '#004D40', 'text': 'White' },
    'A100': { 'color': '#A7FFEB', 'text': 'Black' },
    'A200': { 'color': '#64FFDA', 'text': 'Black' },
    'A400': { 'color': '#1DE9B6', 'text': 'Black' },
    'A700': { 'color': '#00BFA5', 'text': 'Black' }
  },
  'Green': {
    '50': { 'color': '#E8F5E9', 'text': 'Black' },
    '100': { 'color': '#C8E6C9', 'text': 'Black' },
    '200': { 'color': '#A5D6A7', 'text': 'Black' },
    '300': { 'color': '#81C784', 'text': 'Black' },
    '400': { 'color': '#66BB6A', 'text': 'Black' },
    '500': { 'color': '#4CAF50', 'text': 'Black' },
    '600': { 'color': '#43A047', 'text': 'White' },
    '700': { 'color': '#388E3C', 'text': 'White' },
    '800': { 'color': '#2E7D32', 'text': 'White' },
    '900': { 'color': '#1B5E20', 'text': 'White' },
    'A100': { 'color': '#B9F6CA', 'text': 'Black' },
    'A200': { 'color': '#69F0AE', 'text': 'Black' },
    'A400': { 'color': '#00E676', 'text': 'Black' },
    'A700': { 'color': '#00C853', 'text': 'Black' }
  },
  'Light Green': {
    '50': { 'color': '#F1F8E9', 'text': 'Black' },
    '100': { 'color': '#DCEDC8', 'text': 'Black' },
    '200': { 'color': '#C5E1A5', 'text': 'Black' },
    '300': { 'color': '#AED581', 'text': 'Black' },
    '400': { 'color': '#9CCC65', 'text': 'Black' },
    '500': { 'color': '#8BC34A', 'text': 'Black' },
    '600': { 'color': '#7CB342', 'text': 'Black' },
    '700': { 'color': '#689F38', 'text': 'White' },
    '800': { 'color': '#558B2F', 'text': 'White' },
    '900': { 'color': '#33691E', 'text': 'White' },
    'A100': { 'color': '#CCFF90', 'text': 'Black' },
    'A200': { 'color': '#B2FF59', 'text': 'Black' },
    'A400': { 'color': '#76FF03', 'text': 'Black' },
    'A700': { 'color': '#64DD17', 'text': 'Black' }
  },
  'Lime': {
    '50': { 'color': '#F9FBE7', 'text': 'Black' },
    '100': { 'color': '#F0F4C3', 'text': 'Black' },
    '200': { 'color': '#E6EE9C', 'text': 'Black' },
    '300': { 'color': '#DCE775', 'text': 'Black' },
    '400': { 'color': '#D4E157', 'text': 'Black' },
    '500': { 'color': '#CDDC39', 'text': 'Black' },
    '600': { 'color': '#C0CA33', 'text': 'Black' },
    '700': { 'color': '#AFB42B', 'text': 'Black' },
    '800': { 'color': '#9E9D24', 'text': 'Black' },
    '900': { 'color': '#827717', 'text': 'White' },
    'A100': { 'color': '#F4FF81', 'text': 'Black' },
    'A200': { 'color': '#EEFF41', 'text': 'Black' },
    'A400': { 'color': '#C6FF00', 'text': 'Black' },
    'A700': { 'color': '#AEEA00', 'text': 'Black' }
  },
  'Yellow': {
    '50': { 'color': '#FFFDE7', 'text': 'Black' },
    '100': { 'color': '#FFF9C4', 'text': 'Black' },
    '200': { 'color': '#FFF59D', 'text': 'Black' },
    '300': { 'color': '#FFF176', 'text': 'Black' },
    '400': { 'color': '#FFEE58', 'text': 'Black' },
    '500': { 'color': '#FFEB3B', 'text': 'Black' },
    '600': { 'color': '#FDD835', 'text': 'Black' },
    '700': { 'color': '#FBC02D', 'text': 'Black' },
    '800': { 'color': '#F9A825', 'text': 'Black' },
    '900': { 'color': '#F57F17', 'text': 'Black' },
    'A100': { 'color': '#FFFF8D', 'text': 'Black' },
    'A200': { 'color': '#FFFF00', 'text': 'Black' },
    'A400': { 'color': '#FFEA00', 'text': 'Black' },
    'A700': { 'color': '#FFD600', 'text': 'Black' }
  },
  'Amber': {
    '50': { 'color': '#FFF8E1', 'text': 'Black' },
    '100': { 'color': '#FFECB3', 'text': 'Black' },
    '200': { 'color': '#FFE082', 'text': 'Black' },
    '300': { 'color': '#FFD54F', 'text': 'Black' },
    '400': { 'color': '#FFCA28', 'text': 'Black' },
    '500': { 'color': '#FFC107', 'text': 'Black' },
    '600': { 'color': '#FFB300', 'text': 'Black' },
    '700': { 'color': '#FFA000', 'text': 'Black' },
    '800': { 'color': '#FF8F00', 'text': 'Black' },
    '900': { 'color': '#FF6F00', 'text': 'Black' },
    'A100': { 'color': '#FFE57F', 'text': 'Black' },
    'A200': { 'color': '#FFD740', 'text': 'Black' },
    'A400': { 'color': '#FFC400', 'text': 'Black' },
    'A700': { 'color': '#FFAB00', 'text': 'Black' }
  },
  'Orange': {
    '50': { 'color': '#FFF3E0', 'text': 'Black' },
    '100': { 'color': '#FFE0B2', 'text': 'Black' },
    '200': { 'color': '#FFCC80', 'text': 'Black' },
    '300': { 'color': '#FFB74D', 'text': 'Black' },
    '400': { 'color': '#FFA726', 'text': 'Black' },
    '500': { 'color': '#FF9800', 'text': 'Black' },
    '600': { 'color': '#FB8C00', 'text': 'Black' },
    '700': { 'color': '#F57C00', 'text': 'Black' },
    '800': { 'color': '#EF6C00', 'text': 'White' },
    '900': { 'color': '#E65100', 'text': 'White' },
    'A100': { 'color': '#FFD180', 'text': 'Black' },
    'A200': { 'color': '#FFAB40', 'text': 'Black' },
    'A400': { 'color': '#FF9100', 'text': 'Black' },
    'A700': { 'color': '#FF6D00', 'text': 'Black' }
  },
  'Deep Orange': {
    '50': { 'color': '#FBE9E7', 'text': 'Black' },
    '100': { 'color': '#FFCCBC', 'text': 'Black' },
    '200': { 'color': '#FFAB91', 'text': 'Black' },
    '300': { 'color': '#FF8A65', 'text': 'Black' },
    '400': { 'color': '#FF7043', 'text': 'Black' },
    '500': { 'color': '#FF5722', 'text': 'White' },
    '600': { 'color': '#F4511E', 'text': 'White' },
    '700': { 'color': '#E64A19', 'text': 'White' },
    '800': { 'color': '#D84315', 'text': 'White' },
    '900': { 'color': '#BF360C', 'text': 'White' },
    'A100': { 'color': '#FF9E80', 'text': 'Black' },
    'A200': { 'color': '#FF6E40', 'text': 'Black' },
    'A400': { 'color': '#FF3D00', 'text': 'White' },
    'A700': { 'color': '#DD2C00', 'text': 'White' }
  },
  'Brown': {
    '50': { 'color': '#EFEBE9', 'text': 'Black' },
    '100': { 'color': '#D7CCC8', 'text': 'Black' },
    '200': { 'color': '#BCAAA4', 'text': 'Black' },
    '300': { 'color': '#A1887F', 'text': 'White' },
    '400': { 'color': '#8D6E63', 'text': 'White' },
    '500': { 'color': '#795548', 'text': 'White' },
    '600': { 'color': '#6D4C41', 'text': 'White' },
    '700': { 'color': '#5D4037', 'text': 'White' },
    '800': { 'color': '#4E342E', 'text': 'White' },
    '900': { 'color': '#3E2723', 'text': 'White' }
  },
  'Grey': {
    '50': { 'color': '#FAFAFA', 'text': 'Black' },
    '100': { 'color': '#F5F5F5', 'text': 'Black' },
    '200': { 'color': '#EEEEEE', 'text': 'Black' },
    '300': { 'color': '#E0E0E0', 'text': 'Black' },
    '400': { 'color': '#BDBDBD', 'text': 'Black' },
    '500': { 'color': '#9E9E9E', 'text': 'Black' },
    '600': { 'color': '#757575', 'text': 'White' },
    '700': { 'color': '#616161', 'text': 'White' },
    '800': { 'color': '#424242', 'text': 'White' },
    '900': { 'color': '#212121', 'text': 'White' }
  },
  'Blue Grey': {
    '50': { 'color': '#ECEFF1', 'text': 'Black' },
    '100': { 'color': '#CFD8DC', 'text': 'Black' },
    '200': { 'color': '#B0BEC5', 'text': 'Black' },
    '300': { 'color': '#90A4AE', 'text': 'Black' },
    '400': { 'color': '#78909C', 'text': 'White' },
    '500': { 'color': '#607D8B', 'text': 'White' },
    '600': { 'color': '#546E7A', 'text': 'White' },
    '700': { 'color': '#455A64', 'text': 'White' },
    '800': { 'color': '#37474F', 'text': 'White' },
    '900': { 'color': '#263238', 'text': 'White' }
  },

  'Black': {
    '500': { 'color': '#000000', 'text': 'White' },
    'text': { 'color': 'rgba(0,0,0,0.87)', 'text': 'White' },

    'Primary': { 'color': 'rgba(0,0,0,0.87)', 'text': 'White' },
    'Secondary': { 'color': 'rgba(0,0,0,0.54)', 'text': 'White' },
    'Icons': { 'color': 'rgba(0,0,0,0.38)', 'text': 'White' },
    'Disabled': { 'color': 'rgba(0,0,0,0.38)', 'text': 'White' },
    'Hint': { 'color': 'rgba(0,0,0,0.38)', 'text': 'White' },
    'Dividers': { 'color': 'rgba(0,0,0,0.12)', 'text': 'White' }
  },

  'White': {
    '500': { 'color': '#ffffff', 'text': 'Black' },
    'text': { 'color': '#ffffff', 'text': 'White' },

    'Primary': { 'color': '#ffffff', 'text': 'Black' },
    'Secondary': { 'color': 'rgba(255,255,255,0.7)', 'text': 'Black' },
    'Icons': { 'color': 'rgba(255,255,255,0.5)', 'text': 'Black' },
    'Disabled': { 'color': 'rgba(255,255,255,0.5)', 'text': 'Black' },
    'Hint': { 'color': 'rgba(255,255,255,0.5)', 'text': 'Black' },
    'Dividers': { 'color': 'rgba(255,255,255,0.12)', 'text': 'Black' }
  },
}
